packages/consumption/src/modules/requests/incoming/IncomingRequestsController.ts

```diff
@@ -245,6 +272,150 @@ export class IncomingRequestsController extends ConsumptionBaseController {
         }
     }
 
+    private validateKeyUniquenessOfRelationshipAttributesWithinRequest(
+        items: (RequestItem | RequestItemGroup)[],
+        params: (DecideRequestItemParametersJSON | DecideRequestItemGroupParametersJSON)[]
+    ) {
+        const fragmentsOfMustBeAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+        const fragmentsOfAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+
+        for (let i = 0; i < params.length; i++) {
+            const item = items[i];
+            const decideItemParams = params[i];
+
+            if (item instanceof RequestItemGroup) {
+                const fragmentsOfMustBeAcceptedItemsOfGroup = this.extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(item);
+                if (fragmentsOfMustBeAcceptedItemsOfGroup) {
+                    fragmentsOfMustBeAcceptedItemsOfRequest.push(...fragmentsOfMustBeAcceptedItemsOfGroup);
+                }
+
+                const fragmentsOfAcceptedItemsOfGroup = this.extractRelationshipAttributeFragmentsFromAcceptedItemsOfGroup(
+                    item,
+                    decideItemParams as DecideRequestItemGroupParametersJSON
+                );
+                if (fragmentsOfAcceptedItemsOfGroup) {
+                    fragmentsOfAcceptedItemsOfRequest.push(...fragmentsOfAcceptedItemsOfGroup);
+                }
+            } else {
+                const fragmentOfMustBeAcceptedRequestItem = this.extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(item);
+                if (fragmentOfMustBeAcceptedRequestItem) {
+                    fragmentsOfMustBeAcceptedItemsOfRequest.push(fragmentOfMustBeAcceptedRequestItem);
+                }
+
+                const fragmentOfAcceptedRequestItem = this.extractRelationshipAttributeFragmentFromAcceptedRequestItem(item, decideItemParams as DecideRequestItemParametersJSON);
+                if (fragmentOfAcceptedRequestItem) {
+                    fragmentsOfAcceptedItemsOfRequest.push(fragmentOfAcceptedRequestItem);
+                }
+            }
+        }
+
+        if (IncomingRequestsController.containsDuplicateRelationshipAttributeFragments(fragmentsOfMustBeAcceptedItemsOfRequest)) {
+            throw ConsumptionCoreErrors.requests.violatedKeyUniquenessOfRelationshipAttributes(
```

Comment:
Do you mean that the entire Request should be automatically rejected if the Request is faulty in the sense that it contains several `mustBeAccepted` RequestItems, which would lead to the creation of RelationshipAttributes with the same key, owner and value type? Perhaps this discussion goes beyond the scope of the PR for the moment.

As you have already worked a lot with automatically processed Requests @Milena-Czierlinski, what do you think about it in principle?